go/types/struct.go

```diff
@@ -195,13 +210,45 @@ func EscapeStructField(input string) string {
 		buf.WriteString(hs)
 		return buf.String()
 	}
+	output := EscapeFields(input, encode)
+	verifyFieldName(output)
```

Comment:
Isn't the output guaranteed to be valid at this point, assuming `EscapeFields` works properly? If so there's no need for this check.